java/ql/src/experimental/Security/CWE/CWE-489/EJBMain.ql

```diff
@@ -0,0 +1,30 @@
+/**
+ * @name Main Method in Enterprise Java Bean
+ * @description Jave EE applications with a main method.
+ * @kind problem
+ * @id java/main-method-in-enterprise-bean
```

Comment:
Why did you add a  separate query for J2EE? Couldn't this fit into the "web component" category of the other query?

Reply:
In a typical 3-tier architecture, servlets and Spring/Struts actions are in the presentation tier while session beans of EJB are in the application/business tier. Therefore enterprise beans are not web components in general.

Reply:
I see. Thanks for the explanation.